scripts/module/keybinder.nut

```diff
@@ -0,0 +1,54 @@
+keys <- {
+  "a" : 0x41,
+    "b" : 0x42,
+    "c" : 0x43,
+    "d" : 0x44,
+    "e" : 0x45,
+    "f" : 0x46,
+    "g" : 0x47,
+    "h" : 0x48,
+    "i" : 0x49,
+    "j" : 0x4A,
+    "k" : 0x4B,
+    "l" : 0x4C
+    }; // FIXME: Complete the table
+
+
+binds <- array(100, null);
+
+function onPlayerCommand( player, cmd, text ) {
+  cmd = cmd.tolower();
+  if (cmd == "b") {
+    text = text.tolower();
+    local bindTable = {};
+    local textArr = split(text, " ");
+    if (textArr.len() < 2) {
+      MessagePlayer("Usage: /" + cmd + " <key> <command>", player);
+    } else if (!(textArr[0] in keys)) {
+      MessagePlayer("Invalid key!", player);
+    }
+    if (binds[player.ID]) {
+      bindTable = binds[player.ID];
+    }
+    bindTable.rawset(keys[textArr[0]], text.slice(1));
+    binds[player.ID] = bindTable;
+  } else if (cmd == "help") {
+    MessagePlayer("/b", player);
+  }
+}
+
+function onKeyDown( player, id ) {
+  if (binds[player.ID]) {
+    if (binds[player.ID][id]) {
+      local cmdstring = binds[player.ID][id];
+      local cmd = split(cmdstring, " ")[0];
+      getRootTable()["onPlayerCommand"].call(this, cmd, cmdstring.splice(cmd.len()));
```

Comment:
nit: This is a bit ugly in my opinion

Why not:
```
local cmd_end = cmdstring.find(" ");
if (cmd_end == null) {
  cmd_end = cmdstring.len();
}
local cmd = cmdstring.splice(0, cmd_end);
local args = cmdstring.splice(cmd_end + 1);
```

(might contain bugs)

Reply:
Not just a bit IMO. Thanks for providing a much more elegant solution, I must've been half asleep writing this atrocity.

Reply:
As you seem to use such code multiple times, it might be worth to introduce a `partition` function for strings.

*I'm not sure, but there might also be one in vc-mp already? I couldn't find one on the wiki or the squirrel docs though - but I didn't really search much.*